src/index.js

```diff
@@ -168,10 +168,19 @@ class Service {
 
   // STILL NEED TO ADD params argument here.
   create(data) {
-    return this.table.insert(data).run()
-      .then(res => Object.assign({
-        id: (data.id ? data.id : res.generated_keys[0])
-      }, data));
+    const idField = this.id;
+    return this.table.insert(data).run().then(function (res) {
+      if (data[idField]) {
+        if (res.errors) {
+          return Promise.reject(new errors.Conflict('Duplicate primary key'));
```

Comment:
No. It is not the only possible error. There could be other errors, eg. if someone just deleted the table or turned off the server or the object wasn't serializable something else went wrong, then the error could be something else.

But according to RethinkDB specs on the `insert` command (or at least how I interpret them), this appears to be the only logical error if things are otherwise stable.

Maybe we could change the string into `Insert error: possibly duplicate primary key` to be more correct.